initsystem/kubelet/initsystem.go

```diff
@@ -0,0 +1,291 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strings"
+
+	"k8s.io/klog"
+	"sigs.k8s.io/etcdadm/apis"
+	"sigs.k8s.io/etcdadm/service"
+	"sigs.k8s.io/yaml"
+)
+
+// InitSystem runs etcd under the kubelet
+type InitSystem struct {
+	desiredConfig *apis.EtcdAdmConfig
+}
+
+// EnableAndStartService enables and starts the etcd service
+func (s *InitSystem) EnableAndStartService(service string) error {
+	cfg := s.desiredConfig
+
+	name := s.name(cfg)
+
+	// We have to precreate the data dir for kubelet
```

Comment:
Not sure I understand. Please expand. Thanks!

Reply:
Looks like this to ensure correct ownership

For xref, kubeadm uses `DirectoryOrCreate` for this volume, but the owner is `kubelet`, and permissions are `755`.

```
  volumes:
  - hostPath:
      path: /etc/kubernetes/pki/etcd
      type: DirectoryOrCreate
    name: etcd-certs
  - hostPath:
      path: /var/lib/etcd
      type: DirectoryOrCreate
    name: etcd-data
```